src/temaWeek6LocalStore/main/Animal.java

```diff
@@ -0,0 +1,69 @@
+package temaWeek6LocalStore.main;
+//subclass of Product class
+
+public class Animal extends Product {
+	
+	private String storageTemp;
+	
+	
+	//	constructor of Animal class
+	public Animal(String name, double price, String validityDate, double weight, String ID, String storageTemp) {
+		super(name,price, validityDate,weight, ID);//calling constructor from superclass
+		this.storageTemp=storageTemp;
+		
+	}
+	
+	//	method for storageTemp
+	public String setStorageTemp(String storageTemp) {
+		System.out.println("Enter storage temperature: ");
+		String input=scan.next();
+		if(!isDouble(input)){
+			System.out.println("Enter a valid temperature!");
+			setStorageTemp(storageTemp);
+		}
+		return storageTemp=input+"°C";
+	}
+	
+	public String getStorageTemp() {
+		return storageTemp;
+	}
+	
+	@Override
+	public String setName(String name) {
+		System.out.println("Enter  type of product:milk/cheese/yogurt: ");
+		name=scan.next();
```

Comment:
does this work?
... I think this works but because of a weird reason... 
SO: you create String name; in localStore
you pass the name reference to this object in the constructor
you call the setName with the same reference to name
you set the value on the parameter which since it's the same reference would automatically be stored on the animal as well

this is a big mess of references. If you realy want to do System.out and scan.next() here, then store the value directly in the attribute of this class "this.name = ... ", make the name protected and you can change the value from the child class...